src/material-experimental/mdc-checkbox/checkbox.html

```diff
@@ -1,6 +1,6 @@
 <div class="mdc-form-field"
      [class.mdc-form-field--align-end]="labelPosition == 'before'">
-  <div #checkbox class="mdc-checkbox">
+  <div #checkbox class="mdc-checkbox" (click)="$event.stopPropagation()">
```

Comment:
ohhh, good catch . I meant to put the `stopPropgation` on a parent of the label 🤦 😂.

I'm surprised this passes the tests 😕

Reply:
Looks like we have at least two options

```
<!-- option A: put the click handler on the form field-->
<div class="mdc-form-field" (click)="$event.stopPropagation()... >
 ...
</div>
```

```
<!-- option B: wrap label in a div with display: contents; -->
<div style="display: contents;" (click)="$event.stopPropagation()">
  <label #label [for]="inputId">
    <ng-content></ng-content>
  </label
</div>
```

option A looks a little cleaner, but it breaks the bubbling. Click events won't bubble up from the mdc-checkbox anymore. That leaves us with option B, WDYT? Is the bubbling a requirement?

Reply:
Have you checked that moving the input to a parent doesn't re-introduce the issue that the PR is trying to resolve? I don't mind either way as long as we don't regress on something else.

Reply:
> moving the input to a parent
Not sure I understand, but both option A and B produce the correctly accessibility tree. They both remove the extraneous node corresponding to the `<label/>`.

I tested the bubbling with option B, and here is the order that the click event flows to

1. click on label
2. label -> parent of label (`<div style="display: contents" (click)="$event.stopPropagation()">`)-> input -> .mdc-checkbox -> ... -> document

That's because the `display: contents` removes the div from the render tree but keeps it in the dom. It seems like the browser does not consider the div for the a11y tree. I think normally div's are not included in the a11y tree unless they have added attributes.

Reply:
not sure if the `display: contents;` is really necessary

Reply:
Sorry, I meant the listener, not the input. It seems like `display: contents` might cause some accessibility issues in Safari https://caniuse.com/css-display-contents.

Reply:
okay, how about we try using a `<span/>` without the `display: contents;`? If that works, then we don't need to worry about support for display: contents.